### DISCUSSION TITLE
[Q&A] Azure OpenAI 暂不支持  TTS & STT 相关模型

### DISCUSSION THREAD

**Discussion by yanxiyue** [lobehub/lobehub#483] [General Question] [answered]
### 💻 系统环境 | Operating System

Macos

### 🌐 浏览器 | Browser

Chrome

### 🐛 问题描述 | Bug Description
开发模式下，
语音朗读不可用，显示服务器遇到问题。 服务端日志里提示
```
 ⨯ node_modules/openai/error.mjs (48:19) @ APIError.generate
 ⨯ 404 Resource not found
```

复现方法参考：https://jam.dev/c/b844c92f-11a7-452a-b9c5-234be4cece2a

### 🚦 期望结果 | Expected Behavior

语音朗读可用

### 📷 复现步骤 | Recurrence Steps

1. bun run dev
2. 在bot回复的对话框旁边点击更多，语音朗读。

参考：https://jam.dev/c/b844c92f-11a7-452a-b9c5-234be4cece2a

### 📝 补充信息 | Additional Information

打印看了服务端的日志看了一下，访问的url是http://localhost:3010/api/openai/tts，但在tts没有route.ts

[a comment by an automated account is not shown]

**Comment by canisminor1990** [comment]
这是 openai route 的位置 https://github.com/lobehub/lobe-chat/tree/main/src%2Fapp%2Fapi%2Fopenai%2Ftts

**Comment by canisminor1990** [comment]
> 这是 openai route 的位置 https://github.com/lobehub/lobe-chat/tree/main/src%2Fapp%2Fapi%2Fopenai%2Ftts

可在里面 log 你的入参和相关返回以获得更准确的错误定位

**Comment by yanxiyue** [comment, marked as answer]
> > 这是 openai route 的位置 https://github.com/lobehub/lobe-chat/tree/main/src%2Fapp%2Fapi%2Fopenai%2Ftts
> 
> 可在里面 log 你的入参和相关返回以获得更准确的错误定位

感谢快速回复，我看错代码了。 应该是我使用的是azure的原因，没有部署语音的模型。

**Comment by yanxiyue** [comment]
在azure上没找到对应openai tts的模型，先更换为openai的key了，更换之后可以使用。

**Reply by arvinxx** [reply]
Azure现在应该也没有上1106出的新模型。估计要等一段时间了

**Reply by yanxiyue** [reply]
1106上了，我已经部署上了。 加拿大东部区域

**Reply by arvinxx** [reply]
> 1106上了，我已经部署上了。 加拿大东部区域

vision model 也有了？

**Reply by yanxiyue** [reply]
vision的还没有
